server/src-lib/Hasura/Server/Query.hs

```diff
@@ -292,7 +292,10 @@ getQueryAccessMode q = (fromMaybe Q.ReadOnly) <$> getQueryAccessMode' q
     getQueryAccessMode' (RQV1 q') =
       case q' of
         RQSelect _ -> pure Nothing
-        RQCount _ -> pure Nothing
+        RQCount _  -> pure Nothing
+        RQGetInconsistentMetadata _     -> pure Nothing
+        RQExportMetadata _              -> pure Nothing
+        RQDumpInternalState _           -> pure Nothing
         RQRunSql RunSQL {rTxAccessMode} -> pure $ Just rTxAccessMode
         RQBulk qs -> foldM reconcileAccessModeWith Nothing (zip [0 :: Integer ..] qs)
```

Comment:
Use `indexedFoldM` from `RQL.Types.Error` so that you don't have to worry about constructing the path. So, it'll be
`withPathk "args" $ indexedFoldM ...`